src/Sulu/Bundle/MediaBundle/Controller/MediaStreamController.php

```diff
@@ -197,32 +199,33 @@ private function createBinaryFileResponse(
 
     /**
      * @param int $id
-     * @param int $version
+     * @param int|null $version
      *
      * @return FileVersion|null
      *
      * @throws FileVersionNotFoundException
      */
     protected function getFileVersion($id, $version)
     {
-        /** @var MediaInterface $mediaEntity */
-        $mediaEntity = $this->mediaRepository->findMediaByIdForRendering($id, null);
+        /** @var MediaInterface|null $mediaEntity */
+        $mediaEntity = $this->mediaRepository->findMediaByIdForRendering($id, null, $version);
 
         if (!$mediaEntity) {
             return null;
         }
 
+        /** @var File|null $file */
         $file = $mediaEntity->getFiles()[0];
```

Comment:
this breaks when `getFiles()` returns an empty array

```suggestion
        $file = $mediaEntity->getFiles()[0] ?? null;
```

Reply:
`getFiles` is an **ArrayCollection**, the index 0 can be retrieved with `getFiles()->get(0)`, so in the case it does not exists, it returns `null`

Reply:
This is expected we avoid coupling to Collection specific classes in case as we move step by step to a Hexagonial Architecture.